src/api/auth/auth.cpp

```diff
@@ -392,7 +392,6 @@ error::Error Authenticator::WithToken(AuthenticatedAction action) {
 		if (err != error::NoError) {
 			// Should never fail. We rely on the signal heavily so let's fail
 			// hard if it does.
-			action(expected::unexpected(err));
```

Comment:
The test that I wrote verifies this fix.

The other fix below is harder to trigger as we would need somehow a hook in `WithToken` to make the DBus inaccessible _after_ we pass this first part. One hack could be to make a test friend class that could manipulate `watching_token_signal_`, but I regarded that as too ugly to put in production code.

I think it is not worth the effort.

Reply:
Agreed!